src/pudl/analysis/mcoe.py

```diff
@@ -276,13 +281,15 @@ def capacity_factor(pudl_out, min_cap_fact=0, max_cap_fact=1.5):
                                           'report_date',
                                           'generator_id',
                                           'capacity_mw']]
-    gen_eia923 = pudl_out.gen_eia923()[['plant_id_eia',
-                                        'report_date',
-                                        'generator_id',
-                                        'net_generation_mwh']]
+
+    gen = pudl_out.gen_eia923()
+    if pudl_out.fill_net_gen:
+        gen = pudl_out.gen_allocated()
```

Comment:
Would it make sense to move this conditional logic into a helper function in `pudl_out` that would check the status of `fill_net_gen` and return the corresponding dataframe?

Reply:
that is a good idea!

Reply:
hey @rousik I integrated this logic into `pudl_out` as you suggested! I did a little bit of additional column management such that the columns in output of `pudl_out.gen_eia923()` is the same whether `fill_net_gen` is True or False. let me know what you think!